tests/setup.sh

```diff
@@ -0,0 +1,296 @@
+      # unlikely if cri-dockerd wasn't already installed, it is possible.
+      cri_dockerd_service_error=""
+      systemctl is-active --quiet cri-docker.socket || cri_dockerd_service_error=$?
+      if [ -n "${cri_dockerd_service_error}" ]; then
+        # Download systemd service files
+        cri_dockerd_version="v$(cri-dockerd --version 2>&1 | awk '{print $2}')"
+        systemd_service_files_url="https://raw.githubusercontent.com/Mirantis/cri-dockerd/${cri_dockerd_version}/packaging/systemd"
+        cd ${workspace}
+        ${CURL} "${systemd_service_files_url}"/cri-docker.service -o cri-docker.service
+        ${CURL} "${systemd_service_files_url}"/cri-docker.socket -o cri-docker.socket
+        cd -
+        install "${workspace}"/cri-docker.{service,socket} /etc/systemd/system
+        sed -i -e 's,/usr/bin/cri-dockerd,/usr/local/bin/cri-dockerd,' /etc/systemd/system/cri-docker.service
+        systemctl daemon-reload
+        systemctl enable --now cri-docker.socket
+        systemctl is-active --quiet cri-dockerd.socket || die "failed to set up cri-dockerd systemd service" 1
+        echo "Enabled cri-dockerd.socket systemd service."
+      fi
+
+      # Install container-network-plugins.
+      echo "Installing container-network-plugins..."
+      containernetworking_tag=$(get_latest_version_tag "containernetworking" "plugins")
+      cni_plugin_version="${containernetworking_tag}"
+      cni_plugin_tar="cni-plugins-linux-amd64-${cni_plugin_version}.tgz"
+      cni_plugin_install_dir="/opt/cni/bin"
+      cd "${workspace}"
+      ${CURL} -O "https://github.com/containernetworking/plugins/releases/download/${cni_plugin_version}/${cni_plugin_tar}"
+      mkdir -p "${cni_plugin_install_dir}"
+      tar -xf "${cni_plugin_tar}" -C "${cni_plugin_install_dir}"
+      cd - >/dev/null
+      echo "Installed container-network-plugins."
+
+      echo "Installed prerequisites for $os-$arch."
+      ;;
+      # TODO: Needs implementation.
+    "arm64" | "aarch64")
+      die "the 'install_prerequisites' option is not implemented for ${os}-${arch}" 1
+      ;;
+    *)
+      die "the 'install_prerequisites' option does not support the arch ${arch} for OS ${os}" 1
+      ;;
+    esac
+    ;;
+  "Darwin")
+    case "$arch" in
+    # TODO: Needs implementation.
+    "x86_64")
+      die "the 'install_prerequisites' option is not implemented for ${os}-${arch}" 1
+      ;;
+      # TODO: Needs implementation.
+    "arm64")
+      die "the 'install_prerequisites' option is not implemented for ${os}-${arch}" 1
+      ;;
+    *)
+      die "the 'install_prerequisites' option does not support the arch ${arch} for OS ${os}" 1
+      ;;
+    esac
+    ;;
+  *)
+    die "the 'install_prerequisites' option does not support the OS ${os}" 1
+    ;;
+  esac
+}
+
+# Installing minikube and kubectl from hosted binaries because the nix unstable channel
+# versions are usually outdated.
```

Comment:
This is not true, unstable has the latest versions...

Reply:
Oh. I messed up. I forgot to run `nix-channel --update`

Reply:
I'll move back to using nix.